File: lession1/JavaWithPattern/VisitorJavaWithPatternFramework/src/main/java/Visitor2.java
/**
 * Created by Administrator on 2016/12/1.
 */
public class Visitor2 implements Visitor {

    public void view(Element1 element1)
    {
        System.out.println("visitor2 view element1");
    }

    public void view(Element2 element2)
    {
        System.out.println("visitor2 view element2");
    }
}

File: lession1/CSharpDynamic/VisitorExtension/Visitor2.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using VisitorElement;

namespace VisitorExtension
{
    public class Visitor2
    {
        public void View(Element1 element)
        {
            Console.WriteLine("visitor2 view element1");
        }

        public void View(Element2 element)
        {
            Console.WriteLine("visitor2 view element2");
        }
    }
}
